### PR TITLE
Possible docker deployment approach

### PR DESCRIPTION
**Pull request by triblondon**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+WORKDIR .
+ENV HOME="/"
+
+# If you have native dependencies, you'll need extra tools. Also remove any cached files from the installs
+# RUN apk add --no-cache make gcc g++ python
+
+ADD ./package.json ./
+
+# If don't you need npm, use a base tag
+RUN npm install --production
```
Comment (DavidBruant) [inline Dockerfile]: You also want to add 
`ENV NODE_ENV production`
Some packages act differently (and some emit kind warnings on what not to do in prod; `express-session` comes to mind)

Comment (JakeChampion) [inline Dockerfile]: Heroku do set it in the container, it would be better to set it ourselves

**Comment by JakeChampion** [comment]
What's the state of this PR? Should we resolve the conflicts and merge or close it?

**Comment by triblondon** [comment]
Still want to figure out the best approach for this, but if you want to make some process with it, feel free.  My focus will be on Map and then IntersectionObserver until I've shipped those

**Comment by triblondon** [comment]
Struggling with issues like `grunt compatgen` which needs the runtime environment to run.  For the moment we will use docker just for deployment, so I'll modify this PR to limit itself to deployment.
Start with a git clean -fxd?

**Comment by triblondon** [comment]
Hi Rowan - in Jake's absence can you look over this?  Jake and I decided to use docker just for deployment on the polyfill service.  I can build the image and run it locally, and I think I have the correct tasks configured for deployment, but I get this output from Heroku:

```
$ npm run deploy
... snip ...
The push refers to a repository [registry.heroku.com/ft-polyfill-service-qa/web]
0d48b43af4d4: Preparing 
6d64942ab833: Preparing 
976b0b9ce8a3: Preparing 
538e81ce30d6: Preparing 
ea74409b057b: Preparing 
c56b7dabbc7a: Waiting 
no basic auth credentials
```

There don't seem to be any public docs on using Heroku's docker registry for deployment so I assume you guys found out how to do this direct from Heroku somehow, or else are just smarter than me.  Anyway feel free to deploy (`npm run deploy` will deploy to QA) and if it works please let me know how to configure auth!

**Comment by JakeChampion** [comment]
The origami build service should have documentation on how to deploy a
docker image to herokus registry, you need to be a part of the beta and
then register via the docker Cli herokus registry

**Comment by rowanmanning** [comment]
I can take a look. Also yep the [Build Service documentation](https://github.com/Financial-Times/origami-build-service#deployment) outlines how to do this. @triblondon: I'll send you Heroku's docker article via Slack in case you don't have access.

**Comment by triblondon** [comment]
Thanks.  I tried that and got this:

```
Successfully built c15e9da43fcf
The push refers to a repository [registry.heroku.com/ft-polyfill-service-qa/web]
cda391d7472d: Pushed 
e4d18a9393e5: Pushed 
08a4cceb4026: Pushed 
538e81ce30d6: Pushed 
ea74409b057b: Pushed 
c56b7dabbc7a: Pushed 
received unexpected HTTP status: 500 Internal Server Error
```

Could you try it?

**Comment by rowanmanning** [comment]
It deployed but I'm missing Fastly credentials, so `deployvcl:qa` failed. The Docker deploy worked though!

**Comment by triblondon** [comment]
OK, I'm signed up to the beta now and deployment works perfectly.  Since the haikro method is broken for me for some reason, I'm merging so I can resume deploying.

**Pull request merged**